src/asynk/async_queue.rs

```diff
@@ -46,13 +54,44 @@ where
     }
 
     pub async fn insert_task(&mut self, task: &dyn AsyncRunnable) -> Result<u64, AsyncQueueError> {
-        let json_task = serde_json::to_value(task).unwrap();
+        let metadata = serde_json::to_value(task).unwrap();
         let task_type = task.task_type();
 
-        self.execute_one(INSERT_TASK_QUERY, &[&json_task, &task_type])
+        self.execute_one(INSERT_TASK_QUERY, &[&metadata, &task_type])
             .await
     }
-
+    pub async fn update_task_state(
+        &mut self,
+        uuid: Uuid,
+        state: &str,
+        updated_at: NaiveDateTime,
+    ) -> Result<u64, AsyncQueueError> {
+        self.execute_one(UPDATE_TASK_STATE_QUERY, &[&state, &updated_at, &uuid])
+            .await
+    }
+    pub async fn remove_all_tasks(&mut self) -> Result<u64, AsyncQueueError> {
+        self.execute_one(REMOVE_ALL_TASK_QUERY, &[]).await
```

Comment:
this query may fail if the number of removed records is more than 1

Reply:
@ayrat555 i am sure that query will not fail because execute_one calls 

https://docs.rs/tokio-postgres/0.7.0/tokio_postgres/struct.Client.html#method.execute

that do not panic! if two or n records are deleted,  execute method will return as u64 2 or n,  that is then number of rows affected.

Reply:
my comment was related to this check - https://github.com/ayrat555/fang/pull/24/files#diff-24272200c270b7446465f4378bfa6c16bd4a7f88837d283e84b24114d0ec3404L79